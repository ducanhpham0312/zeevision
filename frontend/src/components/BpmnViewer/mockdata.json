{
  "primary": {
    "bpmnString": "test-bpmn-xml-string",
    "navigated": true
  },
  "secondary": {
    "bpmnString": "test-bpmn-xml-string",
    "navigated": false
  },
  "moneyLoan": {
    "bpmnString": "PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPGJwbW46ZGVmaW5pdGlvbnMgeG1sbnM6YnBtbj0iaHR0cDovL3d3dy5vbWcub3JnL3NwZWMvQlBNTi8yMDEwMDUyNC9NT0RFTCIgeG1sbnM6YnBtbmRpPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9CUE1OLzIwMTAwNTI0L0RJIiB4bWxuczpkYz0iaHR0cDovL3d3dy5vbWcub3JnL3NwZWMvREQvMjAxMDA1MjQvREMiIHhtbG5zOnhzaT0iaHR0cDovL3d3dy53My5vcmcvMjAwMS9YTUxTY2hlbWEtaW5zdGFuY2UiIHhtbG5zOmRpPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9ERC8yMDEwMDUyNC9ESSIgeG1sbnM6emVlYmU9Imh0dHA6Ly9jYW11bmRhLm9yZy9zY2hlbWEvemVlYmUvMS4wIiB4bWxuczptb2RlbGVyPSJodHRwOi8vY2FtdW5kYS5vcmcvc2NoZW1hL21vZGVsZXIvMS4wIiBpZD0iRGVmaW5pdGlvbnNfMGhtdnZqMCIgdGFyZ2V0TmFtZXNwYWNlPSJodHRwOi8vYnBtbi5pby9zY2hlbWEvYnBtbiIgZXhwb3J0ZXI9IkNhbXVuZGEgTW9kZWxlciIgZXhwb3J0ZXJWZXJzaW9uPSI1LjEzLjAiIG1vZGVsZXI6ZXhlY3V0aW9uUGxhdGZvcm09IkNhbXVuZGEgQ2xvdWQiIG1vZGVsZXI6ZXhlY3V0aW9uUGxhdGZvcm1WZXJzaW9uPSI4LjIuMCI+CiAgPGJwbW46cHJvY2VzcyBpZD0ibW9uZXktbG9hbiIgaXNFeGVjdXRhYmxlPSJ0cnVlIj4KICAgIDxicG1uOnN0YXJ0RXZlbnQgaWQ9IlN0YXJ0RXZlbnRfMSIgbmFtZT0iTmVlZGVkIHRvIGxvYW4gbW9uZXkiPgogICAgICA8YnBtbjpvdXRnb2luZz5GbG93XzAzZWkxYjQ8L2JwbW46b3V0Z29pbmc+CiAgICA8L2JwbW46c3RhcnRFdmVudD4KICAgIDxicG1uOmV4Y2x1c2l2ZUdhdGV3YXkgaWQ9IkdhdGV3YXlfMHd2ZW8wYiIgbmFtZT0iQ2hlY2sgY3JlZGl0IiBkZWZhdWx0PSJGbG93XzBrd2s2Y2giPgogICAgICA8YnBtbjppbmNvbWluZz5GbG93XzAzZWkxYjQ8L2JwbW46aW5jb21pbmc+CiAgICAgIDxicG1uOm91dGdvaW5nPkZsb3dfMGt3azZjaDwvYnBtbjpvdXRnb2luZz4KICAgICAgPGJwbW46b3V0Z29pbmc+Rmxvd18xc3czZ3B0PC9icG1uOm91dGdvaW5nPgogICAgPC9icG1uOmV4Y2x1c2l2ZUdhdGV3YXk+CiAgICA8YnBtbjplbmRFdmVudCBpZD0iRXZlbnRfMTlmajN5OSIgbmFtZT0iRW5kIj4KICAgICAgPGJwbW46aW5jb21pbmc+Rmxvd18xZTB6NHZjPC9icG1uOmluY29taW5nPgogICAgICA8YnBtbjppbmNvbWluZz5GbG93XzAybGtkdjQ8L2JwbW46aW5jb21pbmc+CiAgICA8L2JwbW46ZW5kRXZlbnQ+CiAgICA8YnBtbjpzZXJ2aWNlVGFzayBpZD0ic2VuZC1hY2NlcHRhbmNlLWxldHRlciIgbmFtZT0iU2VuZCBhY2NlcHRhbmNlIGxldHRlciI+CiAgICAgIDxicG1uOmV4dGVuc2lvbkVsZW1lbnRzPgogICAgICAgIDx6ZWViZTp0YXNrRGVmaW5pdGlvbiB0eXBlPSJzZW5kLWFjY2VwdGFuY2UtbGV0dGVyIiAvPgogICAgICA8L2JwbW46ZXh0ZW5zaW9uRWxlbWVudHM+CiAgICAgIDxicG1uOmluY29taW5nPkZsb3dfMGt3azZjaDwvYnBtbjppbmNvbWluZz4KICAgICAgPGJwbW46b3V0Z29pbmc+Rmxvd18xYnR3M3JuPC9icG1uOm91dGdvaW5nPgogICAgPC9icG1uOnNlcnZpY2VUYXNrPgogICAgPGJwbW46c2VydmljZVRhc2sgaWQ9InRyYW5zZmVyLW1vbmV5IiBuYW1lPSJUcmFuc2ZlciBtb25leSI+CiAgICAgIDxicG1uOmV4dGVuc2lvbkVsZW1lbnRzPgogICAgICAgIDx6ZWViZTp0YXNrRGVmaW5pdGlvbiB0eXBlPSJ0cmFuc2Zlci1tb25leSIgLz4KICAgICAgPC9icG1uOmV4dGVuc2lvbkVsZW1lbnRzPgogICAgICA8YnBtbjppbmNvbWluZz5GbG93XzFidHczcm48L2JwbW46aW5jb21pbmc+CiAgICAgIDxicG1uOm91dGdvaW5nPkZsb3dfMWUwejR2YzwvYnBtbjpvdXRnb2luZz4KICAgIDwvYnBtbjpzZXJ2aWNlVGFzaz4KICAgIDxicG1uOnNlcnZpY2VUYXNrIGlkPSJzZW5kLXJlamVjdGlvbi1sZXR0ZXIiIG5hbWU9IlNlbmQgcmVqZWN0aW9uIGxldHRlciI+CiAgICAgIDxicG1uOmV4dGVuc2lvbkVsZW1lbnRzPgogICAgICAgIDx6ZWViZTp0YXNrRGVmaW5pdGlvbiB0eXBlPSJzZW5kLXJlamVjdGlvbi1sZXR0ZXIiIC8+CiAgICAgIDwvYnBtbjpleHRlbnNpb25FbGVtZW50cz4KICAgICAgPGJwbW46aW5jb21pbmc+Rmxvd18xc3czZ3B0PC9icG1uOmluY29taW5nPgogICAgICA8YnBtbjppbmNvbWluZz5GbG93XzExNnU3YmI8L2JwbW46aW5jb21pbmc+CiAgICAgIDxicG1uOm91dGdvaW5nPkZsb3dfMXRkMnN3MTwvYnBtbjpvdXRnb2luZz4KICAgIDwvYnBtbjpzZXJ2aWNlVGFzaz4KICAgIDxicG1uOmV4Y2x1c2l2ZUdhdGV3YXkgaWQ9IkdhdGV3YXlfMDdqbGM5ayIgbmFtZT0iQ3VzdG9tZXIgcmVjZWl2ZSBsZXR0ZXI/IiBkZWZhdWx0PSJGbG93XzFtNjE4OTgiPgogICAgICA8YnBtbjppbmNvbWluZz5GbG93XzF0ZDJzdzE8L2JwbW46aW5jb21pbmc+CiAgICAgIDxicG1uOm91dGdvaW5nPkZsb3dfMDJsa2R2NDwvYnBtbjpvdXRnb2luZz4KICAgICAgPGJwbW46b3V0Z29pbmc+Rmxvd18xbTYxODk4PC9icG1uOm91dGdvaW5nPgogICAgPC9icG1uOmV4Y2x1c2l2ZUdhdGV3YXk+CiAgICA8YnBtbjppbnRlcm1lZGlhdGVDYXRjaEV2ZW50IGlkPSJFdmVudF8wOXZldzRtIiBuYW1lPSJXYWl0IDIgbWludXRlcyI+CiAgICAgIDxicG1uOmluY29taW5nPkZsb3dfMW02MTg5ODwvYnBtbjppbmNvbWluZz4KICAgICAgPGJwbW46b3V0Z29pbmc+Rmxvd18xMTZ1N2JiPC9icG1uOm91dGdvaW5nPgogICAgICA8YnBtbjp0aW1lckV2ZW50RGVmaW5pdGlvbiBpZD0iVGltZXJFdmVudERlZmluaXRpb25fMWsyeTNjZCI+CiAgICAgICAgPGJwbW46dGltZUR1cmF0aW9uIHhzaTp0eXBlPSJicG1uOnRGb3JtYWxFeHByZXNzaW9uIj5QVDEyMFM8L2JwbW46dGltZUR1cmF0aW9uPgogICAgICA8L2JwbW46dGltZXJFdmVudERlZmluaXRpb24+CiAgICA8L2JwbW46aW50ZXJtZWRpYXRlQ2F0Y2hFdmVudD4KICAgIDxicG1uOnNlcXVlbmNlRmxvdyBpZD0iRmxvd18wM2VpMWI0IiBzb3VyY2VSZWY9IlN0YXJ0RXZlbnRfMSIgdGFyZ2V0UmVmPSJHYXRld2F5XzB3dmVvMGIiIC8+CiAgICA8YnBtbjpzZXF1ZW5jZUZsb3cgaWQ9IkZsb3dfMGt3azZjaCIgc291cmNlUmVmPSJHYXRld2F5XzB3dmVvMGIiIHRhcmdldFJlZj0ic2VuZC1hY2NlcHRhbmNlLWxldHRlciIgLz4KICAgIDxicG1uOnNlcXVlbmNlRmxvdyBpZD0iRmxvd18xc3czZ3B0IiBzb3VyY2VSZWY9IkdhdGV3YXlfMHd2ZW8wYiIgdGFyZ2V0UmVmPSJzZW5kLXJlamVjdGlvbi1sZXR0ZXIiPgogICAgICA8YnBtbjpjb25kaXRpb25FeHByZXNzaW9uIHhzaTp0eXBlPSJicG1uOnRGb3JtYWxFeHByZXNzaW9uIj49ZGVidCAmZ3Q7IDEwMDA8L2JwbW46Y29uZGl0aW9uRXhwcmVzc2lvbj4KICAgIDwvYnBtbjpzZXF1ZW5jZUZsb3c+CiAgICA8YnBtbjpzZXF1ZW5jZUZsb3cgaWQ9IkZsb3dfMWUwejR2YyIgc291cmNlUmVmPSJ0cmFuc2Zlci1tb25leSIgdGFyZ2V0UmVmPSJFdmVudF8xOWZqM3k5IiAvPgogICAgPGJwbW46c2VxdWVuY2VGbG93IGlkPSJGbG93XzAybGtkdjQiIHNvdXJjZVJlZj0iR2F0ZXdheV8wN2psYzlrIiB0YXJnZXRSZWY9IkV2ZW50XzE5ZmozeTkiPgogICAgICA8YnBtbjpjb25kaXRpb25FeHByZXNzaW9uIHhzaTp0eXBlPSJicG1uOnRGb3JtYWxFeHByZXNzaW9uIj49cmVjZWl2ZWQgPSB0cnVlPC9icG1uOmNvbmRpdGlvbkV4cHJlc3Npb24+CiAgICA8L2JwbW46c2VxdWVuY2VGbG93PgogICAgPGJwbW46c2VxdWVuY2VGbG93IGlkPSJGbG93XzFidHczcm4iIHNvdXJjZVJlZj0ic2VuZC1hY2NlcHRhbmNlLWxldHRlciIgdGFyZ2V0UmVmPSJ0cmFuc2Zlci1tb25leSIgLz4KICAgIDxicG1uOnNlcXVlbmNlRmxvdyBpZD0iRmxvd18xMTZ1N2JiIiBzb3VyY2VSZWY9IkV2ZW50XzA5dmV3NG0iIHRhcmdldFJlZj0ic2VuZC1yZWplY3Rpb24tbGV0dGVyIiAvPgogICAgPGJwbW46c2VxdWVuY2VGbG93IGlkPSJGbG93XzFtNjE4OTgiIHNvdXJjZVJlZj0iR2F0ZXdheV8wN2psYzlrIiB0YXJnZXRSZWY9IkV2ZW50XzA5dmV3NG0iIC8+CiAgICA8YnBtbjpzZXF1ZW5jZUZsb3cgaWQ9IkZsb3dfMXRkMnN3MSIgc291cmNlUmVmPSJzZW5kLXJlamVjdGlvbi1sZXR0ZXIiIHRhcmdldFJlZj0iR2F0ZXdheV8wN2psYzlrIiAvPgogIDwvYnBtbjpwcm9jZXNzPgogIDxicG1uZGk6QlBNTkRpYWdyYW0gaWQ9IkJQTU5EaWFncmFtXzEiPgogICAgPGJwbW5kaTpCUE1OUGxhbmUgaWQ9IkJQTU5QbGFuZV8xIiBicG1uRWxlbWVudD0ibW9uZXktbG9hbiI+CiAgICAgIDxicG1uZGk6QlBNTlNoYXBlIGlkPSJCUE1OU2hhcGVfMHNrbHlyciIgYnBtbkVsZW1lbnQ9IlN0YXJ0RXZlbnRfMSI+CiAgICAgICAgPGRjOkJvdW5kcyB4PSIxNzIiIHk9IjE4MiIgd2lkdGg9IjM2IiBoZWlnaHQ9IjM2IiAvPgogICAgICAgIDxicG1uZGk6QlBNTkxhYmVsPgogICAgICAgICAgPGRjOkJvdW5kcyB4PSIxNTMiIHk9IjIyNSIgd2lkdGg9Ijc1IiBoZWlnaHQ9IjI3IiAvPgogICAgICAgIDwvYnBtbmRpOkJQTU5MYWJlbD4KICAgICAgPC9icG1uZGk6QlBNTlNoYXBlPgogICAgICA8YnBtbmRpOkJQTU5TaGFwZSBpZD0iQlBNTlNoYXBlXzAydmthbXgiIGJwbW5FbGVtZW50PSJHYXRld2F5XzB3dmVvMGIiIGlzTWFya2VyVmlzaWJsZT0idHJ1ZSI+CiAgICAgICAgPGRjOkJvdW5kcyB4PSIyOTUiIHk9IjE3NSIgd2lkdGg9IjUwIiBoZWlnaHQ9IjUwIiAvPgogICAgICAgIDxicG1uZGk6QlBNTkxhYmVsPgogICAgICAgICAgPGRjOkJvdW5kcyB4PSIzNTUiIHk9IjE5MyIgd2lkdGg9IjYyIiBoZWlnaHQ9IjE0IiAvPgogICAgICAgIDwvYnBtbmRpOkJQTU5MYWJlbD4KICAgICAgPC9icG1uZGk6QlBNTlNoYXBlPgogICAgICA8YnBtbmRpOkJQTU5TaGFwZSBpZD0iQlBNTlNoYXBlXzBtYTIwbzAiIGJwbW5FbGVtZW50PSJFdmVudF8xOWZqM3k5Ij4KICAgICAgICA8ZGM6Qm91bmRzIHg9Ijg5MiIgeT0iMTgyIiB3aWR0aD0iMzYiIGhlaWdodD0iMzYiIC8+CiAgICAgICAgPGJwbW5kaTpCUE1OTGFiZWw+CiAgICAgICAgICA8ZGM6Qm91bmRzIHg9Ijg2MiIgeT0iMTkzIiB3aWR0aD0iMjAiIGhlaWdodD0iMTQiIC8+CiAgICAgICAgPC9icG1uZGk6QlBNTkxhYmVsPgogICAgICA8L2JwbW5kaTpCUE1OU2hhcGU+CiAgICAgIDxicG1uZGk6QlBNTlNoYXBlIGlkPSJCUE1OU2hhcGVfMHZtZ3l5diIgYnBtbkVsZW1lbnQ9InNlbmQtYWNjZXB0YW5jZS1sZXR0ZXIiPgogICAgICAgIDxkYzpCb3VuZHMgeD0iNDUwIiB5PSI4MCIgd2lkdGg9IjEwMCIgaGVpZ2h0PSI4MCIgLz4KICAgICAgPC9icG1uZGk6QlBNTlNoYXBlPgogICAgICA8YnBtbmRpOkJQTU5TaGFwZSBpZD0iQlBNTlNoYXBlXzFhajIyYTciIGJwbW5FbGVtZW50PSJ0cmFuc2Zlci1tb25leSI+CiAgICAgICAgPGRjOkJvdW5kcyB4PSI2NjAiIHk9IjgwIiB3aWR0aD0iMTAwIiBoZWlnaHQ9IjgwIiAvPgogICAgICA8L2JwbW5kaTpCUE1OU2hhcGU+CiAgICAgIDxicG1uZGk6QlBNTlNoYXBlIGlkPSJCUE1OU2hhcGVfMTBndzM0NyIgYnBtbkVsZW1lbnQ9InNlbmQtcmVqZWN0aW9uLWxldHRlciI+CiAgICAgICAgPGRjOkJvdW5kcyB4PSI0NDAiIHk9IjI1MCIgd2lkdGg9IjEwMCIgaGVpZ2h0PSI4MCIgLz4KICAgICAgPC9icG1uZGk6QlBNTlNoYXBlPgogICAgICA8YnBtbmRpOkJQTU5TaGFwZSBpZD0iQlBNTlNoYXBlXzBrMW42aGYiIGJwbW5FbGVtZW50PSJFdmVudF8wOXZldzRtIj4KICAgICAgICA8ZGM6Qm91bmRzIHg9IjYxMiIgeT0iMzgyIiB3aWR0aD0iMzYiIGhlaWdodD0iMzYiIC8+CiAgICAgICAgPGJwbW5kaTpCUE1OTGFiZWw+CiAgICAgICAgICA8ZGM6Qm91bmRzIHg9IjU5NCIgeT0iNDI1IiB3aWR0aD0iNzMiIGhlaWdodD0iMTQiIC8+CiAgICAgICAgPC9icG1uZGk6QlBNTkxhYmVsPgogICAgICA8L2JwbW5kaTpCUE1OU2hhcGU+CiAgICAgIDxicG1uZGk6QlBNTlNoYXBlIGlkPSJCUE1OU2hhcGVfMGRzbWxjMSIgYnBtbkVsZW1lbnQ9IkdhdGV3YXlfMDdqbGM5ayIgaXNNYXJrZXJWaXNpYmxlPSJ0cnVlIj4KICAgICAgICA8ZGM6Qm91bmRzIHg9Ijc1NSIgeT0iMjY1IiB3aWR0aD0iNTAiIGhlaWdodD0iNTAiIC8+CiAgICAgICAgPGJwbW5kaTpCUE1OTGFiZWw+CiAgICAgICAgICA8ZGM6Qm91bmRzIHg9IjczNyIgeT0iMjM1IiB3aWR0aD0iODciIGhlaWdodD0iMjciIC8+CiAgICAgICAgPC9icG1uZGk6QlBNTkxhYmVsPgogICAgICA8L2JwbW5kaTpCUE1OU2hhcGU+CiAgICAgIDxicG1uZGk6QlBNTkVkZ2UgaWQ9IkJQTU5FZGdlXzBsbzNxY2EiIGJwbW5FbGVtZW50PSJGbG93XzAzZWkxYjQiPgogICAgICAgIDxkaTp3YXlwb2ludCB4PSIyMDgiIHk9IjIwMCIgLz4KICAgICAgICA8ZGk6d2F5cG9pbnQgeD0iMjk1IiB5PSIyMDAiIC8+CiAgICAgIDwvYnBtbmRpOkJQTU5FZGdlPgogICAgICA8YnBtbmRpOkJQTU5FZGdlIGlkPSJCUE1ORWRnZV8wZDBpemE0IiBicG1uRWxlbWVudD0iRmxvd18wa3drNmNoIj4KICAgICAgICA8ZGk6d2F5cG9pbnQgeD0iMzIwIiB5PSIxNzUiIC8+CiAgICAgICAgPGRpOndheXBvaW50IHg9IjMyMCIgeT0iMTIwIiAvPgogICAgICAgIDxkaTp3YXlwb2ludCB4PSI0NTAiIHk9IjEyMCIgLz4KICAgICAgPC9icG1uZGk6QlBNTkVkZ2U+CiAgICAgIDxicG1uZGk6QlBNTkVkZ2UgaWQ9IkJQTU5FZGdlXzF2YTN3MDgiIGJwbW5FbGVtZW50PSJGbG93XzFzdzNncHQiPgogICAgICAgIDxkaTp3YXlwb2ludCB4PSIzMjAiIHk9IjIyNSIgLz4KICAgICAgICA8ZGk6d2F5cG9pbnQgeD0iMzIwIiB5PSIyOTAiIC8+CiAgICAgICAgPGRpOndheXBvaW50IHg9IjQ0MCIgeT0iMjkwIiAvPgogICAgICA8L2JwbW5kaTpCUE1ORWRnZT4KICAgICAgPGJwbW5kaTpCUE1ORWRnZSBpZD0iQlBNTkVkZ2VfMGEycjNzOSIgYnBtbkVsZW1lbnQ9IkZsb3dfMWUwejR2YyI+CiAgICAgICAgPGRpOndheXBvaW50IHg9Ijc2MCIgeT0iMTIwIiAvPgogICAgICAgIDxkaTp3YXlwb2ludCB4PSI5MTAiIHk9IjEyMCIgLz4KICAgICAgICA8ZGk6d2F5cG9pbnQgeD0iOTEwIiB5PSIxODIiIC8+CiAgICAgIDwvYnBtbmRpOkJQTU5FZGdlPgogICAgICA8YnBtbmRpOkJQTU5FZGdlIGlkPSJCUE1ORWRnZV8xeW93MHAyIiBicG1uRWxlbWVudD0iRmxvd18wMmxrZHY0Ij4KICAgICAgICA8ZGk6d2F5cG9pbnQgeD0iODA1IiB5PSIyOTAiIC8+CiAgICAgICAgPGRpOndheXBvaW50IHg9IjkxMCIgeT0iMjkwIiAvPgogICAgICAgIDxkaTp3YXlwb2ludCB4PSI5MTAiIHk9IjIxOCIgLz4KICAgICAgPC9icG1uZGk6QlBNTkVkZ2U+CiAgICAgIDxicG1uZGk6QlBNTkVkZ2UgaWQ9IkJQTU5FZGdlXzFyNXh1ODEiIGJwbW5FbGVtZW50PSJGbG93XzFidHczcm4iPgogICAgICAgIDxkaTp3YXlwb2ludCB4PSI1NTAiIHk9IjEyMCIgLz4KICAgICAgICA8ZGk6d2F5cG9pbnQgeD0iNjYwIiB5PSIxMjAiIC8+CiAgICAgIDwvYnBtbmRpOkJQTU5FZGdlPgogICAgICA8YnBtbmRpOkJQTU5FZGdlIGlkPSJCUE1ORWRnZV8waHM2bmh5IiBicG1uRWxlbWVudD0iRmxvd18xMTZ1N2JiIj4KICAgICAgICA8ZGk6d2F5cG9pbnQgeD0iNjEyIiB5PSI0MDAiIC8+CiAgICAgICAgPGRpOndheXBvaW50IHg9IjQ5MCIgeT0iNDAwIiAvPgogICAgICAgIDxkaTp3YXlwb2ludCB4PSI0OTAiIHk9IjMzMCIgLz4KICAgICAgPC9icG1uZGk6QlBNTkVkZ2U+CiAgICAgIDxicG1uZGk6QlBNTkVkZ2UgaWQ9IkJQTU5FZGdlXzE2MWJnamQiIGJwbW5FbGVtZW50PSJGbG93XzFtNjE4OTgiPgogICAgICAgIDxkaTp3YXlwb2ludCB4PSI3ODAiIHk9IjMxNSIgLz4KICAgICAgICA8ZGk6d2F5cG9pbnQgeD0iNzgwIiB5PSI0MDAiIC8+CiAgICAgICAgPGRpOndheXBvaW50IHg9IjY0OCIgeT0iNDAwIiAvPgogICAgICA8L2JwbW5kaTpCUE1ORWRnZT4KICAgICAgPGJwbW5kaTpCUE1ORWRnZSBpZD0iRmxvd18xdGQyc3cxX2RpIiBicG1uRWxlbWVudD0iRmxvd18xdGQyc3cxIj4KICAgICAgICA8ZGk6d2F5cG9pbnQgeD0iNTQwIiB5PSIyOTAiIC8+CiAgICAgICAgPGRpOndheXBvaW50IHg9Ijc1NSIgeT0iMjkwIiAvPgogICAgICA8L2JwbW5kaTpCUE1ORWRnZT4KICAgIDwvYnBtbmRpOkJQTU5QbGFuZT4KICA8L2JwbW5kaTpCUE1ORGlhZ3JhbT4KPC9icG1uOmRlZmluaXRpb25zPgo=",
    "colorOptions": [
      {
        "elementId": "send-acceptance-letter",
        "intent": "ELEMENT_ACTIVATED"
      },
      {
        "elementId": "send-rejection-letter",
        "intent": "ELEMENT_COMPLETED"
      },
      {
        "elementId": "transfer-money",
        "intent": "ELEMENT_TERMINATED"
      },
      {
        "elementId": "Flow_03ei1b4",
        "intent": "SEQUENCE_FLOW_TAKEN"
      },
      {
        "elementId": "Gateway_0wveo0b",
        "intent": "INCIDENT_CREATED"
      }
    ],
    "navigated": true,
    "control": true
  }
}
