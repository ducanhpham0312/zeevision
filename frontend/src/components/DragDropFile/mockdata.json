{
  "file": {
    "content": "Test File DragnDrop",
    "name": "test.bpmn",
    "type": "text/plain"
  }
}
