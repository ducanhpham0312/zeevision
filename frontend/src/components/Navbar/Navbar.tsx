import { Button } from "../Button";
import { useNavigate, useLocation } from "react-router-dom";

const NavigationPath: { name: string; path: string }[] = [
  {
    name: "PROCESSES",
    path: "/processes",
  },
  {
    name: "INSTANCES",
    path: "/instances",
  },
  {
    name: "INCIDENTS",
    path: "/incidents",
  },
  {
    name: "JOBS",
    path: "/jobs",
  },
  {
    name: "MESSAGES",
    path: "/messages",
  },
  {
    name: "ERRORS",
    path: "/errors",
  },
];

export function Navbar() {
  const navigate = useNavigate();
  const location = useLocation();

  return (
    <header className="fixed z-50 flex w-full items-center gap-7 border-b border-black/20 bg-background px-5">
      <div>
        <div>NAVBAR</div>
      </div>
      <div className="h-10 w-0 border-l border-black/30" />
      <div className="flex gap-3">
        {NavigationPath.map((nav) => (
          <Button
            active={location.pathname.includes(nav.path)}
            size="large"
            onClick={() => navigate(nav.path)}
            key={nav.path}
          >
            {nav.name}
          </Button>
        ))}
      </div>
    </header>
  );
}
