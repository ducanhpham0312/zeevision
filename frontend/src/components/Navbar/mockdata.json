[
  {
    "name": "PROCESSES",
    "path": "/processes"
  },
  {
    "name": "INSTANCES",
    "path": "/instances"
  },
  {
    "name": "INCIDENTS",
    "path": "/incidents"
  },
  {
    "name": "JOBS",
    "path": "/jobs"
  },
  {
    "name": "MESSAGES",
    "path": "/messages"
  },
  {
    "name": "ERRORS",
    "path": "/errors"
  }
]
