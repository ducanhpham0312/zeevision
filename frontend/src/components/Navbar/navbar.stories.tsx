import type { Meta, StoryObj } from "@storybook/react";
import { BrowserRouter } from "react-router-dom";
import { Navbar } from ".";

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
  title: "Assets/Navbar",
  component: Navbar,
  parameters: {
    layout: "padded",
  },
  tags: ["autodocs"],
  argTypes: {},
} satisfies Meta<typeof Navbar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: () => (
    <BrowserRouter>
      <div className="flex items-center justify-center">
        <Navbar />
      </div>
    </BrowserRouter>
  ),
};
