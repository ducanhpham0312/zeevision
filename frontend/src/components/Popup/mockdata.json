{
  "openPopup": {
    "open": true,
    "title": "Test Title",
    "children": ["Test child 1", "Test child 2"]
  },
  "closedPopup": {
    "open": false,
    "title": "Test Title",
    "children": ["Test child 1", "Test child 2"]
  },
  "primary": {
    "open": true,
    "title": "Deploy a Process",
    "children": []
  },
  "scrollable": {
    "open": true,
    "title": "Deploy a Process",
    "children": [
      "Cras mattis consectetur purus sit amet fermentum.",
      "Cras justo odio, dapibus ac facilisis in, egestas eget quam.",
      "Morbi leo risus, porta ac consectetur ac, vestibulum at eros.",
      "Praesent commodo cursus magna, vel scelerisque nisl consectetur et."
    ]
  }
}
