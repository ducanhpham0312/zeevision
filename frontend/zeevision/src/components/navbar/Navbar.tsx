import { styled } from "@mui/system";

function Navbar() {
  return (
    <NavbarComponent>
      NAVABR
    </NavbarComponent>
  )
}

const NavbarComponent = styled("header")(({}) => `
  display: flex;

`)

export default Navbar;
