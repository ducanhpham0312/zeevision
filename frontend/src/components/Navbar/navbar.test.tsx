import React from "react";
import "@testing-library/jest-dom";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Navbar } from "./Navbar";
import * as NavigationPath from "./mockdata.json";

// Create a custom render function to include the router context
const customRender = (ui: React.ReactElement, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);

  return render(ui, { wrapper: BrowserRouter });
};

describe("Navbar Component", () => {
  it("renders correctly", () => {
    const { asFragment } = customRender(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders all navigation items", () => {
    customRender(<Navbar />);
    NavigationPath.forEach((navItem) => {
      expect(screen.getByText(navItem.name)).toBeInTheDocument();
    });
  });

  it("navigates to correct path on button click", () => {
    customRender(<Navbar />);
    NavigationPath.forEach((navItem) => {
      fireEvent.click(screen.getByText(navItem.name));
      expect(window.location.pathname).toBe(navItem.path);
    });
  });

  it("highlights the active navigation item", () => {
    NavigationPath.forEach(async (navItem) => {
      customRender(<Navbar />, { route: navItem.path });
      const allButtons = screen.getAllByRole("button", { name: navItem.name });
      // Test the active one button
      const activeButton = allButtons[0];

      // Wait for the expected style to be applied for the active button
      await waitFor(() => {
        const activeStyleDiv = activeButton.querySelector("div");
        expect(activeStyleDiv).not.toBeNull();
        expect(activeStyleDiv).toHaveStyle("height: 4px");
      });
    });
  });
});
