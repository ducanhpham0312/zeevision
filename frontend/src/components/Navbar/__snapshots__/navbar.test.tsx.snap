// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component renders correctly 1`] = `
<DocumentFragment>
  <header
    class="fixed z-50 flex w-full items-center gap-5 border-b border-black/20 bg-background px-5"
  >
    <div>
      <div>
        ZEEVISION
      </div>
    </div>
    <div
      class="h-10 w-0 border-l border-black/30"
    />
    <div
      class="flex"
    >
      <button
        class="relative box-border text-accent hover:bg-hover active:bg-active rounded px-4 py-4"
      >
        <div
          class="transition-width absolute bottom-0 left-0 w-full transform rounded bg-accent duration-100"
          style="height: 0px;"
        />
        <span>
          <p>
            PROCESSES
          </p>
        </span>
      </button>
      <button
        class="relative box-border text-accent hover:bg-hover active:bg-active rounded px-4 py-4"
      >
        <div
          class="transition-width absolute bottom-0 left-0 w-full transform rounded bg-accent duration-100"
          style="height: 0px;"
        />
        <span>
          <p>
            INSTANCES
          </p>
        </span>
      </button>
      <button
        class="relative box-border text-accent hover:bg-hover active:bg-active rounded px-4 py-4"
      >
        <div
          class="transition-width absolute bottom-0 left-0 w-full transform rounded bg-accent duration-100"
          style="height: 0px;"
        />
        <span>
          <p>
            INCIDENTS
          </p>
        </span>
      </button>
      <button
        class="relative box-border text-accent hover:bg-hover active:bg-active rounded px-4 py-4"
      >
        <div
          class="transition-width absolute bottom-0 left-0 w-full transform rounded bg-accent duration-100"
          style="height: 0px;"
        />
        <span>
          <p>
            JOBS
          </p>
        </span>
      </button>
      <button
        class="relative box-border text-accent hover:bg-hover active:bg-active rounded px-4 py-4"
      >
        <div
          class="transition-width absolute bottom-0 left-0 w-full transform rounded bg-accent duration-100"
          style="height: 0px;"
        />
        <span>
          <p>
            MESSAGES
          </p>
        </span>
      </button>
      <button
        class="relative box-border text-accent hover:bg-hover active:bg-active rounded px-4 py-4"
      >
        <div
          class="transition-width absolute bottom-0 left-0 w-full transform rounded bg-accent duration-100"
          style="height: 0px;"
        />
        <span>
          <p>
            ERRORS
          </p>
        </span>
      </button>
    </div>
  </header>
</DocumentFragment>
`;
